jetty-start/src/main/java/org/eclipse/jetty/start/StartArgs.java

```diff
@@ -331,21 +331,23 @@ public void dumpEnvironment(PrintStream out)
 
     public void dumpJvmArgs(PrintStream out)
     {
-        if (jvmArgs.isEmpty())
+        if (jvmArgSources.isEmpty())
             return;
 
         out.println();
         out.println("Forked JVM Arguments:");
         out.println("---------------------");
 
-        for (String jvmArgKey : jvmArgs)
-        {
-            String value = System.getProperty(jvmArgKey);
-            if (value != null)
-                out.printf(" %s = %s%n", jvmArgKey, value);
-            else
-                out.printf(" %s%n", jvmArgKey);
-        }
+        jvmArgSources.forEach((key, sourceRef) ->
+            {
+                String value = System.getProperty(key);
+                String source = StartLog.isDebugEnabled() ? '(' + sourceRef + ')' : "";
+                if (value != null)
+                    out.printf(" %s = %s %s%n", key, value, source);
+                else
+                    out.printf(" %s %s%n", key, source);
+            }
+        );
```

Comment:
Weird braces and parens alignment, can you make it normal?

Reply:
This passes checkstyle.
I'll give it whirl.

Reply:
Done.